File: Second Seminar/SecondSeminar/src/main/java/com/example/SecondSeminar/member/exception/MemberExceptionType.java
package com.example.SecondSeminar.member.exception;

import com.example.SecondSeminar.common.exception.ExceptionType;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@RequiredArgsConstructor
public enum MemberExceptionType implements ExceptionType {
    NOT_FOUND_MEMBER(HttpStatus.NOT_FOUND, "존재하지 않는 멤버입니다.");

    private final HttpStatus httpStatus;
    private final String message;

    @Override
    public HttpStatus httpStatus() {
        return httpStatus;
    }

    @Override
    public String message() {
        return message;
    }
}

File: Second Seminar/SecondSeminarKotlin/src/main/kotlin/com/example/SecondSeminarKotlin/member/exception/MemberExceptionType.kt
package com.example.SecondSeminarKotlin.member.exception

import com.example.SecondSeminarKotlin.exception.ExceptionType
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatus.NOT_FOUND
import org.springframework.http.HttpStatus.UNAUTHORIZED

enum class MemberExceptionType(
    override val httpStatus: HttpStatus,
    override val message: String
) : ExceptionType {
    NOT_FOUND_MEMBER(NOT_FOUND, "존재하지 않는 멤버입니다."),
    INCORRECT_PASSWORD(UNAUTHORIZED, "비밀번호가 올바르지 않습니다.")

}